src/pybind/mgr/zabbix/module.py

```diff
@@ -0,0 +1,209 @@
+"""
+Zabbix module for ceph-mgr
+
+Collect statistics from Ceph cluster and every X seconds send data to a Zabbix
+server using the zabbix_sender executable.
+"""
+import json
+import subprocess
+from Queue import Queue, Empty
+from mgr_module import MgrModule
+
+
+class ZabbixSender(object):
+    def __init__(self, sender, host, port, log):
+        self.sender = sender
+        self.host = host
+        self.port = port
+        self.log = log
+
+    def send(self, hostname, data):
+        if len(data) == 0:
+            return
+
+        try:
+            cmd = '{0} -z {1} -p {2} -s {3} -vv -i -'.format(self.sender,
+                                                             self.host,
+                                                             self.port,
+                                                             hostname)
+
+            proc = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                                    stdout=subprocess.PIPE)
+
+            for key, value in data.items():
+                proc.stdin.write('{0} ceph.{1} {2}\n'.format(hostname, key,
+                                                             str(value)))
+
+            stdout, stderr = proc.communicate()
```

Comment:
Does zabbix-sender ever block, like if the network goes away or something?  The module serve methods are each in their own thread, so that's usually okay, but if zabbix-sender was stuck during shutdown it would be awkward.

There probably isn't a neat way to make the subprocess killable, just thinking out loud.